modules/src/core/ics23_commitment/merkle.rs

```diff
@@ -24,9 +25,148 @@ pub fn apply_prefix(
     Ok(MerklePath { key_path: result })
 }
 
+impl From<CommitmentRoot> for MerkleRoot {
+    fn from(root: CommitmentRoot) -> Self {
+        Self {
+            hash: root.into_vec(),
+        }
+    }
+}
+
 #[derive(Clone, Debug, PartialEq)]
 pub struct MerkleProof {
-    pub proof: Vec<CommitmentProofBytes>,
+    pub proofs: Vec<CommitmentProof>,
+}
+
+/// Convert to ics23::CommitmentProof
+/// The encoding and decoding shouldn't fail since ics23::CommitmentProof and ibc_proto::ics23::CommitmentProof should be the same
+/// Ref. https://github.com/informalsystems/ibc-rs/issues/853
+impl From<RawMerkleProof> for MerkleProof {
+    fn from(proof: RawMerkleProof) -> Self {
+        let proofs: Vec<CommitmentProof> = proof
+            .proofs
+            .into_iter()
+            .map(|p| {
+                let mut encoded = Vec::new();
+                prost::Message::encode(&p, &mut encoded).unwrap();
+                prost::Message::decode(&*encoded).unwrap()
+            })
+            .collect();
+        Self { proofs }
+    }
+}
+
+impl MerkleProof {
```

Comment:
I'm curious: are the two methods implemented here amenable to simple unit-testing? Intuitively it seems so to me. If yes, we should probably do that (but separate issue/PR is fine, not necessarily here!).

Reply:
Right, I just opened an issue to deal with this. https://github.com/informalsystems/ibc-rs/issues/1706